apiserver/facades/client/application/application_unit_test.go

```diff
@@ -701,6 +702,42 @@ func (s *ApplicationSuite) TestDeployAttachStorage(c *gc.C) {
 	c.Assert(results.Results[2].Error, gc.ErrorMatches, `"volume-baz-0" is not a valid volume tag`)
 }
 
+func (s *ApplicationSuite) TestDeployMinDeploymentVersionTooHigh(c *gc.C) {
+	s.model.modelType = state.ModelTypeCAAS
+	s.backend.charm = &mockCharm{
+		meta: &charm.Meta{
+			Deployment: &charm.Deployment{
+				DeploymentType: "stateless",
```

Comment:
This test is for min version check so we don't need to set these other unrelated deployment attributes